app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/AccountController.scala

```diff
@@ -70,4 +73,26 @@ class AccountController @Inject()(
     }
   }
 
+  def getExistingUndertaking: Action[AnyContent] = escAuthentication.async { implicit request =>
+    implicit val eori: EORI = request.eoriNumber
+    for {
+      undertakingOpt <- escService.retrieveUndertaking(eori)
+    } yield {
+      undertakingOpt match {
+        case Some(undertaking) =>
+          if(isLeadEORI(undertaking, eori))
+          Redirect(routes.AccountController.getAccountPage())  //if logged in as lead EORI, redirect to Account home page
+        else
+          Ok(existingUndertakingPage(undertaking.name, eori)) //if logged in as non-lead EORI, redirect to existing undertaking page
+        case None => Redirect(routes.AccountController.getAccountPage()) //if undertaking not present, then create undertaking
+      }
+    }
+  }
+
+  //checks if the logged in EORI is a lead
+  private  def isLeadEORI(undertaking: Undertaking, eori: EORI): Boolean = {
+    val leadEORI = undertaking.undertakingBusinessEntity.filter(_.leadEORI).headOption.getOrElse(handleMissingSessionData("Missing Lead EORI"))
```

Comment:
Minor - it can make things easier to read to break long statements like this over multiple lines e.g. 

```
val leadEORI = undertaking.undertakingBusinessEntity
  .filter(_.leadEORI)
  .headOption
  .getOrElse(handleMissingSessionData("Missing Lead EORI"))
```

We should revisit our error handling later since it's a shame we're throwing exceptions rather than using an Either (or whatever). Something for another day I think!

Reply:
yeah we can definitely revisit this thing later. Normally i am lazy and depend on scalafmt file to do formatting 🤣 . I can definitely do the formatting